src/type_level_bool.rs

```diff
@@ -0,0 +1,30 @@
+#[derive(Copy, Clone, Debug)]
+pub struct True;
+#[derive(Copy, Clone, Debug)]
+pub struct False;
+
+pub trait ToBool {
+    type Value;
+    fn to_bool(_: Self) -> Self::Value;
+}
+impl ToBool for [(); 0] {
+    type Value = False;
+    fn to_bool(_: Self) -> Self::Value {
+        False
+    }
+}
+impl ToBool for [(); 1] {
+    type Value = True;
+    fn to_bool(_: Self) -> Self::Value {
+        True
+    }
+}
```

Comment:
Wow mapping array types to 

Some small changes please:
- `To` usually implies a fallible operation; `Into` feels more appropriate here.
- Separate these `impl`s with an empty line.